Source/WebCore/platform/graphics/filters/FilterResults.cpp

```diff
@@ -40,8 +42,31 @@ FilterImage* FilterResults::effectResult(FilterEffect& effect) const
     return m_results.get(effect);
 }
 
+size_t FilterResults::memoryCost() const
+{
+    CheckedSize memoryCost;
+
+    for (auto& result : m_results.values()) {
+        memoryCost += result->memoryCost();
+        RELEASE_ASSERT(!memoryCost.hasOverflowed());
+    }
+
+    return memoryCost;
+}
+
+bool FilterResults::canCacheResult(FilterImage& result) const
+{
+    CheckedSize totalMemoryCost = memoryCost() + result.memoryCost();
+    if (totalMemoryCost.hasOverflowed())
+        return false;
```

Comment:
Earlier we will crash if the two individual `memoryCost()` calls overflow while they're computing their costs. Do you want to do the same here too, rather than return false?

Reply:
I think this different from `memoryCost()`. FilterImage and FilterResults should stay always within limits. So every time we call `memoryCost()` we can assert/crash we do not overflow. But here we are checking whether adding a new FilterImage to FilterResults will make us within the limits or not.